plugins/microchip/switchtec-nvme-device.c

```diff
@@ -199,7 +199,12 @@ int pax_nvme_ns_rescan(int fd)
 
 int pax_nvme_submit_passthru(int fd, int ioctl_cmd, struct nvme_passthru_cmd *cmd)
 {
-	fprintf(stderr, "%s not implemented.\n", __FUNCTION__);
+	if(ioctl_cmd == (int)NVME_IOCTL_ADMIN_CMD) {
+		return pax_nvme_submit_admin_passthru(fd, cmd);
+	} else {
+		fprintf(stderr, "%s only supports NVME_IOCTL_ADMIN_CMD.\n",
+			__FUNCTION__);
+	}
```

Comment:
Normally we do parameter check and error handling at the very beginning. Then the logic would be more clear. Like

``` 
        if (ioctl_cmd != (int)NVME_IOCTL_ADMIN_CMD)) {
                do error handling;
                return -1;
        }

        do normal processing;
        return 0;
```